sdk/storage/Azure.Storage.Internal.Avro/src/AvroParser.cs

```diff
@@ -0,0 +1,388 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+#pragma warning disable SA1402 // File may only contain a single type
+
+namespace Azure.Storage.Internal.Avro
+{
+    internal static class AvroParser
+    {
+        public static async Task<byte[]> ReadFixedBytesAsync(
+            Stream stream,
+            int length,
```

Comment:
Worst case avro predicts is `bytes` that's as long as `long` can define. Would it be a problem for QQ ?
![image](https://user-images.githubusercontent.com/61715331/80015295-584ac280-8486-11ea-9285-5cc7b50a8fbc.png)

Reply:
I believe the max size of an array in .Net is max int, so we can't return more bytes than that from this method.  QQ records are limited to 4 MB, so I don't think this will be an issue.

https://docs.microsoft.com/en-us/dotnet/api/system.array.length?view=netstandard-2.0

Reply:
This is internal, so I'm not worried about getting into another "we need to expose longs" problem. Especially since every stream read method in .NET uses int for the count.

Reply:
4MB sounds like safe limit.

Would returning Iterable<byte> ease "we need to expose longs" effort here? (should that ever happen).

Reply:
Maybe we could address this in the future, if this situation comes up.  I'd prefer not to change this now, as it would require a significant refactor of the parser (which is currently working with Quick Query and Change Feed).

Reply:
Can we at least leave a comment expressing this, so if anyone picks it up and enhances it, they can keep that in mind?

Reply:
I added a comment mentioning this.